server/src-lib/Hasura/RQL/GBoolExp.hs

```diff
@@ -436,19 +436,62 @@ mkColCompExp qual lhsCol = mkCompExp (mkQCol lhsCol)
         sqlAll = foldr (S.BEBin S.AndOp) (S.BELit True)
         pgTypeToAnnType = S.TypeAnn . T.pack . show
 
-getColExpDeps :: QualifiedTable -> AnnBoolExpFld a -> [SchemaDependency]
+hasStaticExp :: OpExpG PartialSQLExp -> Bool
+hasStaticExp = \case
+  ACast casts  -> any hasStaticExp $ concat $ M.elems casts
+
+  AEQ _ val    -> isStaticValue val
+  ANE _ val    -> isStaticValue val
+
+  AIN val          -> isStaticValue val
+  ANIN val         -> isStaticValue val
+  AGT val          -> isStaticValue val
+  ALT val          -> isStaticValue val
+  AGTE val         -> isStaticValue val
+  ALTE val         -> isStaticValue val
+
+  ALIKE val        -> isStaticValue val
+  ANLIKE val       -> isStaticValue val
+  AILIKE val       -> isStaticValue val
+  ANILIKE val      -> isStaticValue val
+
+  ASIMILAR val     -> isStaticValue val
+  ANSIMILAR val    -> isStaticValue val
+
+  AContains val    -> isStaticValue val
+  AContainedIn val -> isStaticValue val
+  AHasKey val      -> isStaticValue val
+  AHasKeysAny val  -> isStaticValue val
+  AHasKeysAll val  -> isStaticValue val
+
+  ASTContains val   -> isStaticValue val
+  ASTCrosses val    -> isStaticValue val
+  ASTEquals val     -> isStaticValue val
+  ASTIntersects val -> isStaticValue val
+  ASTOverlaps val   -> isStaticValue val
+  ASTTouches val    -> isStaticValue val
+  ASTWithin val     -> isStaticValue val
+  ASTDWithinGeom (DWithinGeomOp r val)     -> any isStaticValue [r, val]
+  ASTDWithinGeog (DWithinGeogOp r val sph) -> any isStaticValue [r, val, sph]
+
+  _                -> False
+
```

Comment:
Is there a better way to achieve this?

Reply:
It’s possible that it could be done more concisely with something like [`Data.Data.Lens`](http://hackage.haskell.org/package/lens-4.17.1/docs/Data-Data-Lens.html). I’m not an expert, so take this with a grain of salt, but if you derive `Data` instances for the necessary types, it could be as simple as writing

```haskell
hasStaticExp :: OpExpG PartialSQLExp -> Bool
hasStaticExp = has (template.filtered isStaticValue)
```

which would be pretty neat.